app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -139,6 +140,7 @@ class BrowserTabViewModel(
     private val contentBlocking: ContentBlocking,
     private val networkLeaderboardDao: NetworkLeaderboardDao,
     private val bookmarksDao: BookmarksDao,
+    private val bookmarksRepository: BookmarksRepository,
```

Comment:
why do we need `BookmarksDao` and `BookmarksRepository`. If the Repo is already using Dao, can we just use one the two?

Reply:
Having both `BookmarksDao` and `BookmarksRepository` is not ideal. I see it as a first step towards moving away from using DAOs inside a viewmodel and in general for moving away from using entities in viewmodel (i.e BookmarkEntity).
I can remove the `BookmarksDao` as part of this PR or a separate one. The change is not big.